transformer_engine/pytorch/attention.py

```diff
@@ -366,7 +368,7 @@ def __init__(
         self.norm_factor = norm_factor
         self.attention_dropout_ctx = attention_dropout_ctx
         self.attention_dropout = attention_dropout
-        self.deterministic = not bool(int(os.getenv("NVTE_ALLOW_NONDETERMINISTIC_ALGO", "1")))
+        self.deterministic = deterministic
```

Comment:
It may be worth adding a check as long as FA 2 is non-deterministic:
```suggestion
        self.deterministic = deterministic
        if self.deterministic:
            assert not _flash_attn_2_available, "Flash Attention 2 does not support deterministic behavior"
```

Reply:
We disable FA and throw a warning in case of deterministic use